lib/asn1/src/asn1ct_check.erl

```diff
@@ -1710,18 +1710,23 @@ check_value(S,#valuedef{pos=Pos,name=Name,type=Type,
     {valueset,
      check_type(S,#typedef{pos=Pos,name=Name,typespec=NewType},NewType)};
 check_value(S, #valuedef{}=V) ->
-    ?dbg("check_value, V: ~p~n",[V0]),
+    ?dbg("check_value, V: ~p~n",[V]),
     case V of
 	#valuedef{checked=true} ->
 	    V;
+	#valuedef{value = [{pv,{simpledefinedvalue,
```

Comment:
Can you explain why there is a special case for a parameterized value here (rewritten to a seqtag record)? Which case does that solve? Is there a test case that catches this case? Seems strange to me, maybe this case should have been taken care of in another function?

Reply:
This solves the test case added in "Add additional value test for SEQUENCE OF SEQUENCE" (
6f9060d )

Since the value is most definitely **not** parameterised, the solution should probably be in the parser however I could not convince myself there isn't an ambiguity in the ASN.1 spec that leads to the problem. I could, however, identify/fix the misrepresentation at the point in this patch.

As an example, if the value was like this then it does not consider seq1 a pv and therefore compiles ok:
`
sos Seq2 ::= {
  seq1 {
    {
      int 10
    }
  },
  nullopt NULL
}
`

One option to move this PR forward is to drop these two commits and I will file a bug on this case.

Reply:
@dotsimon , yes I think we can drop the commit for SEQUENCE OF SEQUENCE" and treat that as a separate bug. I have checked a bit more and I think it should be solved in the parser (if possible). The might be some ambiguities that makes this difficult.

Reply:
Ok I have dropped the 2 commits (test and "fix") for SEQUENCE OF SEQUENCE and force pushed.

ERL-874 is created with the same snipped of ASN.1 that was in the test.